presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java

```diff
@@ -140,11 +140,13 @@ public ConnectorSplitSource getPartitionSplits(ConnectorSession session, Connect
         private final TupleDomain<RaptorColumnHandle> effectivePredicate;
         private final CloseableIterator<ShardNodes> iterator;
 
+        private CompletableFuture<List<ConnectorSplit>> future;
```

Comment:
`ConnectorSplitSource` should not need to be thread safe (from talking to @dain and my brief review of `SourcePartitionedScheduler` and `SqlQueryScheduler`).